src/main/java/com/google/sps/servlets/ProfileBuilderServlet.java

```diff
@@ -0,0 +1,80 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.PersistentUserRepository;
+import com.google.sps.data.User;
+import java.io.IOException;
+import java.lang.NullPointerException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for creating new profiles. */
+@WebServlet("/profile")
+public class ProfileBuilderServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userType = request.getParameter("user-Type");
+
+    if(userType != null) {
+        if(userType.equals("Mentee")){
+            //TODO: add the GoogleUserID
+            String userName = request.getParameter("name-input");
+            User currentUser = new User(userName);
```

Comment:
If userName is null, then the rest of the code in this if block is not used. Consider moving the userName != null check before line 37 to exit this if block early when userName is null.

You can apply the same change to Line 54 for the Mentor else if block.

Reply:
Done